promql/engine.go

```diff
@@ -118,6 +121,20 @@ func (e ErrStorage) Error() string {
 	return e.Err.Error()
 }
 
+func (e ErrTimeExprNotAFloat) Error() string {
+	return fmt.Sprintf("time expression evaluates to %s not one float", string(e))
+}
+
+var (
+	// ErrTimeExprUsesStorage is returned if a time expression
+	// (timestamp/duration) calculation includes vector/range selector.
+	ErrTimeExprUsesStorage = errors.New("time expression must not use storage")
```

Comment:
Would `time expression must not query for series` be a clearer error message to users?